main.js

```diff
@@ -57,26 +57,35 @@ document.addEventListener('DOMContentLoaded', function(e) {
     };
 
     const doDoh = function() {
-        document.getElementById('doh-url-form').classList.add('was-validated');
-
         responseElem.childNodes.forEach(node => node.remove());
+        $loadingModal.modal('show');
+        document.getElementById('do-doh').disabled = true;
 
-        let url = urlInputElem.value;
-        if (!url) {
-            return;
-        }
+        let url = urlInputElem.value || "https://cloudflare-dns.com/dns-query";
         if (corsSwitch.checked) {
             url = cors_proxy + url;
         }
         const method = document.getElementById('doh-method').value || 'POST';
         const qname = document.getElementById('doh-qname').value || '.';
         const qtype = document.getElementById('doh-qtype').value || 'A';
-        $loadingModal.modal('show');
-        document.getElementById('do-doh').disabled = true;
-        const resolver = new doh.DohResolver(url);
-        resolver.query(qname, qtype, method)
-          .then(successFunction)
-          .catch(errorFunction);
+
+        const query = doh.makeQuery(qname, qtype);
+        if (dnssecSwitch.checked) {
+            query.additionals =  [{
+                type: 'OPT',
+                name: '.',
+                udpPayloadSize: 4096,
+                flags: 1 << 15, // DO bit
+                options: []
+            }]
+        }
+        console.log(JSON.stringify(query));
+        console.log();
+        console.log(query.flags & doh.AUTHENTIC_DATA);
+
+        doh.sendDohMsg(query, url, method)
```

Comment:
Not matching the actual signature here which takes 5 params (headers + timeout missing). I can set the headers to null, but not timeout. What's appropriate there?

Reply:
I think it's fine as is. We can add those in later if we feel like it. (JavaScript doesn't care)